json_object.c

```diff
@@ -1265,3 +1265,128 @@ int json_object_equal(struct json_object* jso1, struct json_object* jso2)
 	return 0;
 }
 
+static int json_object_copy_serializer_data(struct json_object *src, struct json_object *dst)
+{
+	/* `src` and `dst` were validated by now ==> no null check for them */
+	if (src->o_type != json_type_double)
```

Comment:
Although we currently only use json_object_userdata_to_json_string for double's, there's no particular reason it couldn't be used for other types as well, and if it's in place we really don't care what type the object is, so please drop this check.